### PR TITLE
Create NobleClient for interacting with noble

### PR DESCRIPTION
**Pull request by rosepuppy**
Run `v4-client-js/examples/noble_example.ts` to test ibc transfers between dydx and noble testnets.

### CODE REVIEW

File: v4-client-js/src/clients/native.ts [hunk 1]
```diff
@@ -988,3 +1017,54 @@ export async function getMarketPrice(
...
+      throw new UserError('wallet is not set. Call connectWallet() first');
+    }
+    const coin = await client.getAccountBalance(globalThis.nobleWallet.address, 'uusdc');
+    return encodeJson(coin);
+  } catch (error) {
+    return wrappedError(error);
+  }
+}
+
+export async function sendNobleIBC(squidPayload: string): Promise<String> {
```
Comment (jaredvu) [inline v4-client-js/src/clients/native.ts:1038]: does this only accept squidPayloads or does it except generic payloads as well?

Comment (rosepuppy) [reply v4-client-js/src/clients/native.ts:1038]: it does specifically have to be a squid IBC payload as I parse the json here with that expectation. Similar to `withdrawToIBC` in this file

File: v4-client-js/src/clients/native.ts [hunk 2]
```diff
@@ -988,3 +1017,66 @@ export async function getMarketPrice(
...
+    const json = JSON.parse(decoded);
+
+    const ibcMsg: EncodeObject = {
+      typeUrl: json.msgTypeUrl, // '/ibc.applications.transfer.v1.MsgTransfer',
+      value: json.msg,
+    };
+    const fee = await client.simulateTransaction([ibcMsg]);
+
+    // take out fee from amount before sweeping
+    ibcMsg.value.token.amount = (parseInt(ibcMsg.value.token.amount, 10) -
```
Comment (ruixhuang) [inline v4-client-js/src/clients/native.ts:1074]: Can we ensure ibcMsg.value.token.amount is positive?  Just to be safe.

Comment (rosepuppy) [reply v4-client-js/src/clients/native.ts:1074]: yes, can do.

**Review by ruixhuang — APPROVED** [review]

**Pull request merged**